src/test/java/testresults/PutTestResultsCertificateNotGenerated.java

```diff
@@ -0,0 +1,160 @@
+        List<JsonPathAlteration> alterationsVehicle = new ArrayList<>(Arrays.asList(alterationSystemNumberVehicle, alterationVinVehicle));
+
+        String alteredBody = GenericData.applyJsonAlterations(postRequestBody, alterationsVehicle);
+        Response response = given().filters(new BasePathFilter())
+                .contentType(ContentType.JSON)
+                .body(alteredBody)
+                .log().method().log().uri().log().body()
+                .post("/vehicles");
+        if (response.getStatusCode() == HttpStatus.SC_UNAUTHORIZED || response.getStatusCode() == HttpStatus.SC_FORBIDDEN) {
+            saveUtils();
+            response = given().filters(new BasePathFilter())
+                    .contentType(ContentType.JSON)
+                    .body(alteredBody)
+                    .log().method().log().uri().log().body()
+                    .post("/vehicles");
+        }
+
+        Assert.assertEquals(HttpStatus.SC_CREATED, response.getStatusCode());
+
+        String testResultRecord = GenericData.readJsonValueFromFile("test-results_payload_10711.json", "$");
+
+        randomTestResultId = UUID.randomUUID().toString();
+        JsonPathAlteration alterationSystemNumberTestResults = new JsonPathAlteration("$.systemNumber", randomSystemNumber, "", "REPLACE");
+        JsonPathAlteration alterationVinTestResults = new JsonPathAlteration("$.vin", randomVin, "", "REPLACE");
+        JsonPathAlteration alterationTestResultIdPost = new JsonPathAlteration("$.testResultId", randomTestResultId, "", "REPLACE");
+
+        List<JsonPathAlteration> alterationsTestResults = new ArrayList<>(Arrays.asList(alterationSystemNumberTestResults, alterationVinTestResults,alterationTestResultIdPost));
+
+        String alteredBodyTestResult = GenericData.applyJsonAlterations(testResultRecord, alterationsTestResults);
+
+        Response responseTestResult = given().filters(new BasePathFilter())
+                .contentType(ContentType.JSON)
+                .body(alteredBodyTestResult)
+                .log().method().log().uri().log().body()
+                .post("/test-results");
+        if (responseTestResult.getStatusCode() == HttpStatus.SC_UNAUTHORIZED || responseTestResult.getStatusCode() == HttpStatus.SC_FORBIDDEN) {
+            saveUtils();
+            responseTestResult = given().filters(new BasePathFilter())
+                    .contentType(ContentType.JSON)
+                    .body(alteredBodyTestResult)
+                    .log().method().log().uri().log().body()
+                    .post("/test-results");
+        }
+        Assert.assertEquals(HttpStatus.SC_CREATED, responseTestResult.getStatusCode());
+    }
+
+    @Steps
+    TestResultsSteps testResultsSteps;
+
+    @Steps
+    VehicleTechnicalRecordsSteps vehicleTechnicalRecordsSteps;
+
+    @TestData
+    public static Collection<Object[]> testData(){
+        return Arrays.asList(new Object[][]{
+                {"fail","cancelled","Annual With Certificate"},
```

Comment:
here you can test also combinations with "abandoned" as test result, which means two more entries in the data provider.

Reply:
Changes done

Reply:
Changes done